package/MDAnalysis/analysis/hole.py

```diff
@@ -1275,15 +1271,23 @@ def _process_orderparameters(self, data):
             raise ValueError(errmsg)
         return q
 
-    def run(self, **kwargs):
+    def run(self, start=None, stop=None, step=None):
```

Comment:
You still need to keep `**kwargs` because the existing API allows you to change the HOLE parameters, see line `hole_kw.update(kwargs)`.

Reply:
We can remove this undocumented functionality, though.

So we keep your code but add a versionchanged that says that all HOLE parameters need to be set with the class constructor HOLEtraj.

Reply:
Should be done in fcf8e23.